jsyntrax/src/main/java/org/atpfivt/jsyntrax/styles/StyleConfig.java

```diff
@@ -0,0 +1,151 @@
+    }
+
+    public StyleConfig(Path style) {
+        this();
+        try {
+            Wini ini = new Wini(style.toFile());
+            if (ini.containsKey("style")) {
+                parseStyleArgs(ini.get("style"));
+            }
+
+            nodeStyles.clear();
+
+            for (Section s : ini.values()) {
+                if (s.getName().equals("style")) {
+                    continue;
+                }
+                // this is custom node style
+                NodeStyle ns = new NodeStyle();
+                ns.name = s.getName();
+                parseNodeStyle(s, ns);
+                nodeStyles.add(ns);
+            }
+        }
+        catch(NoSuchFieldException | IllegalAccessException | IOException e) {}
+    }
+
+    private void parseStyleArgs(Section s) throws NoSuchFieldException, IllegalAccessException {
+        parseField(s, this, "line_width", Integer::parseInt);
+        parseField(s, this, "outline_width", Integer::parseInt);
+        parseField(s, this, "padding", Integer::parseInt);
+        parseField(s, this, "line_color", StyleConfig::colorFromString);
+        parseField(s, this, "max_radius", Integer::parseInt);
+        parseField(s, this, "h_sep", Integer::parseInt);
+        parseField(s, this, "v_sep", Integer::parseInt);
+        parseField(s, this, "arrows", Boolean::parseBoolean);
+        parseField(s, this, "title_pos", (String v) -> v.substring(1, v.length() - 1));
+        parseField(s, this, "bullet_fill", StyleConfig::colorFromString);
+        parseField(s, this, "text_color", StyleConfig::colorFromString);
+        parseField(s, this, "shadow", Boolean::parseBoolean);
+        parseField(s, this, "shadow_fill", StyleConfig::colorFromString);
+        parseField(s, this, "title_font", StyleConfig::fontFromString);
+        parseField(s, this, "scale", Double::parseDouble);
+        parseField(s, this, "transparent", Boolean::parseBoolean);
+    }
+
+    private void parseNodeStyle(Section s, NodeStyle ns)
+            throws NoSuchFieldException, IllegalAccessException {
+        Function<String, Object> patFunc = (String v) -> Pattern.compile(v.substring(1, v.length() - 1));
+        parseField(s, ns, "pattern", patFunc);
+        parseField(s, ns, "shape", (String v) -> v.substring(1, v.length() - 1));
+        parseField(s, ns, "font", StyleConfig::fontFromString);
+        parseField(s, ns, "text_color", StyleConfig::colorFromString);
+        parseField(s, ns, "fill", StyleConfig::colorFromString);
+    }
+
+    private static void parseField(Section s, Object obj,
+                                   String name, Function<String, Object> parser)
+            throws NoSuchFieldException, IllegalAccessException {
+        if (s.containsKey(name)) {
+            obj.getClass().getDeclaredField(name).set(obj,
+                    parser.apply(s.get(name)));
+        }
+    }
+
+    private static Font fontFromString(String txt) {
+        String[] fields = txt.replace("(", "")
```

Comment:
it's better to rewrite this using regexp group matching (https://inponomarev.github.io/java-mipt/slides05/#/_%D1%80%D0%B5%D0%B3%D1%83%D0%BB%D1%8F%D1%80%D0%BD%D1%8B%D0%B5_%D0%B2%D1%8B%D1%80%D0%B0%D0%B6%D0%B5%D0%BD%D0%B8%D1%8F)